agimus_demos_common/agimus_demos_common/happypose_simulation.py

```diff
@@ -4,20 +4,96 @@
 from rclpy.qos import qos_profile_system_default
 from rclpy.executors import ExternalShutdownException
 from vision_msgs.msg import Detection2DArray, Detection2D, ObjectHypothesisWithPose
+from rcl_interfaces.srv import SetParameters
+from rclpy.parameter import Parameter
 
 from tf2_ros import TransformException
 from tf2_ros.buffer import Buffer
 from tf2_ros.transform_listener import TransformListener
 
 import pinocchio
 import numpy as np
+import typing as T
 
 from agimus_controller_ros.ros_utils import (
     transform_msg_to_se3,
     se3_to_pose_msg,
 )
 
 
+def set_transform(
+    node: rclpy.node.Node,
+    object_id: str,
+    camera_name: str,
+    base_name: str,
+    object_pose: T.Union[
+        pinocchio.SE3, tuple[float, float, float, float, float, float, float]
+    ],
+) -> bool:
+    """
+    Set the transform published by the Happypose simulation node. This is the client-side function.
+
+    Args:
+    - node: a ROS node used for ROS communication.
+    - object_pose: desired pose of the object with respect to base_name
+    - camera_name: camera frame name
+    - object_id: name of the object as published by happypose. E.g. tless-obj_000021
+
+    Return:
+    - a boolean that is true if everything the operation was successful.
+    """
+    if isinstance(object_pose, pinocchio.SE3):
+        object_pose = pinocchio.SE3ToXYZQUAT(object_pose)
+
+    assert isinstance(object_pose, (list, tuple)) and len(object_pose) == 7
```

Comment:
correct me if I'm wrong but since `pinocchio.SE3ToXYZQUAT` returns a numpy array I think it should be
```suggestion
    assert isinstance(object_pose, (list, tuple,np.ndarray)) and len(object_pose) == 7
```